messaging/WFLY-13793_disable_legacy_amq1_prefix.adoc

```diff
@@ -0,0 +1,127 @@
+= [WFLY-13793] Allow for a remote jms queue / topic not to use legacy amq1 prefix
+:author:            Chao Wang
+:email:             chaowan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:          messaging,jms
+:idseparator:       -
+
+== Overview
+Currently we can disable the use of legacy amq1 prefixes `jms.queue` and `jms.topic` only at the connection factory level but not for queue or topic defined in the model or using JakartaEE annotations.
+The goal of this proposal is to be able to disable the usage of such prefixes at the destination level for external JMS queues and topics.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-13793[WFLY-13793]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1559[EAP7-1559]
+* https://issues.redhat.com/browse/EAP7-1244[EAP7-1244]
+
+Previous EAP7-1244 exposed the enable1Prefixes configuration on external pooled connection factories, but it does not allow to disable the use of legacy amq1 prefix for queue or topic defined in the model or using JakartaEE annotations.
+
+=== Dev Contacts
+
+* mailto:chaowan@redhat.com[Chao Wang]
+* mailto:ehugonne@redhat.com[Emmanuel Hugonnet]
+
+=== QE Contacts
+
+tborgato@redhat.com
+
+=== Testing By
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
```

Comment:
@soul2zimate IUUC these new attributes have precedence over `messaging-activemq.pooled-connection-factory.remote-artemis:write-attribute(name`: if that's correct can you say it explicitly here?

Reply:
I don't see it as a precedence. Before this RFE, the `enable-amq1-prefix` attribute in the `messaging-activemq.pooled-connection-factory` level doesn't control on the remote queue/topic creation. It always creates remote queue/topic with the `jms.queue` / `jms.topic` prefix. I think you have also illustrated that in your [helloworld-mdb example]( https://github.com/tommaso-borgato/helloworld-mdb). The new attribute added in the destination level `messaging-activemq.external-jms-queue` in this RFE give the ability to configure that.

Reply:
@soul2zimate yes, but there is also the **send/receive phase**: what if, for example, `messaging-activemq.pooled-connection-factory.enable-amq1-prefix=true`  and  `external-jms-queue name.enable-amq1-prefix=false` (or vice-versa)? which of the two setting has precedence?

Reply:
We will honor both: we will create a destination address with the name provided and pass it to the RA. The RA will treat the address with its own configuration (so by default it would append the prefixes). Then of course the broker itself will do some more changes according to its configuration.

Reply:
I think that if the 2 settings don't match we might want to warn the user that his configuration looks inconsistent.

Reply:
@jbertram can you confirm this ?

Reply:
IUUC, in the case that I've depicted, we are creating a queue without prefix e.g. 'myQueue' (honoring just `messaging-activemq.external-jms-queue.enable-amq1-prefix=false`) .... when later sending a message, then both `messaging-activemq.external-jms-queue.enable-amq1-prefix=false` and `messaging-activemq.pooled-connection-factory.enable-amq1-prefix=true` are involved: the first doesn't add anything while the second will add `jms.queue` thus producing a message for queueu `jms.queue.myQueue`

is that correct?

Reply:
If Artemis is configured with auto-create-queue to true, the queue is created, else the message is not delivered (but i think you would get an error as the queue doesn't exist). ActiveMQNonExistentQueueException[errorType=QUEUE_DOES_NOT_EXIST message=AMQ229017: Queue jms.queue.myQueue does not exist]